### DISCUSSION TITLE
Using let_it_be while having tenants in specs

### DISCUSSION THREAD

**Discussion by NBuhinicek** [test-prof/test-prof#231] [Q&A]
Hey guys,

I want to make my specs faster :D So I ended up on a tip to use `let_it_be` for defining some variables that I currently define using a simple `let` but there is a problem.

I'm using the `acts_as_tenant` gem because my app is tenanted and in my `rails_helper.rb` I define 

```ruby
config.before(:each, type: :policy) do
  ActsAsTenant.current_tenant = create(:account)
end
```

And when I put pry into the `let_it_be` method, I stop there before I get into the configured before each block which sets the tenant. And that results with the creation of my `let_it_be` variables throwing errors `Account can not be blank` bcz indead, we haven't yet set the current_tenant :| 

Do you have an idea how to tackle that problem? 

Thanks!

**Comment by palkan** [comment]
Yeah, let_it_be is invoked in the `before(:all)` context, before `before(:each)`.
I suggest using a shared context instead of a global hook:

```ruby
RSpec.shared_context "policy:tenant" do
  let_It_be(:current_tenant) { create(:account) }
  
  before { ActsAsTenant.current_tenant = current_tenant }
end

RSpec.configure do |config|
  config.include_context "policy:tenant", type: :policy
end
```

And then you can refer to this account as `current_tenant` anywhere in your specs.
